csrc/executor.cpp

```diff
@@ -331,6 +331,29 @@ void FusionExecutor::compileFusion(
   lowered_->run();
 
   kir::Kernel* kernel = lowered_->kernel();
+
+  const kir::KernelSummary& kernel_summary = kernel->summary();
+
+  // TODO: this replicates the target GPU version computation from
+  // executor_utils.
+  std::pair<int, int> target_arch;
+  bool compile_to_sass = false;
+  executor_utils::queryTargetGPUVersion(
+      properties, target_arch.first, target_arch.second, compile_to_sass);
+
+  NVF_CHECK(
```

Comment:
Could you move this code down after `getStructuredCode` before nvRTC? Doing so is helpful because, for example, when I am developing Hopper, I don't want to always grab a Hopper machine: sometimes I can just use whatever machine and using `DebugDumpOption::CudaKernel` to check is sufficient for me.

Reply:
Great suggestion. I moved it and verified that the generated kernel is now printed before the error.